ucca/convert.py

```diff
@@ -879,51 +907,54 @@ def from_json(lines, *args, skip_category_mapping=False, by_external_id=False, *
         except KeyError as e:
             raise ValueError("Unit %s appears before its parent, %s" % (tree_id, parent_tree_id)) from e
 
-        tags = []
+        unit_categories = []
         for category in unit.get("categories", ()):
             try:
-                category_name = category.get("name") or category_id_to_name[category["id"]]
+                category_name = category.get("name") or categories[category["id"]]['name']
             except KeyError as e:
                 raise ValueError("Category missing from layer: " + category["id"]) from e
-            tag = category_name_to_edge_tag.get(category_name.replace(" ", ""), category_name)
-            tags.append(tag)
-
-        if not tags:
+            c_tag = category_name_to_edge_tag.get(category_name.replace(" ", ""), category_name.replace(" ", "_"))
+            c_slot = category["slot"] if 'slot' in category else ""
```

Comment:
`category["slot"] if 'slot' in category else ""` can be replaced by `category.get("slot", "")`